spyder/widgets/projects/explorer.py

```diff
@@ -149,12 +151,29 @@ def dropEvent(self, event):
                                        "<br><br>Error message:<br>%s"
                                        ) % (action_str, src,
                                             to_text_string(error)))
+   
+    @Slot()
+    def delete(self, fnames=None):
+        """Delete files"""
+        if fnames is None:
+            fnames = self.get_selected_filenames()
+        multiple = len(fnames) > 1
+        yes_to_all = None
+        for fname in fnames:
+            if fname == self.proxymodel.path_list[0]:
+                self.delete_project.emit()
+            else:
+                yes_to_all = self.delete_file(fname, multiple, yes_to_all)
+                if yes_to_all is not None and not yes_to_all:
+                    # Canceled
+                    break
 
 
 class ProjectExplorerWidget(QWidget):
     """Project Explorer"""
     sig_option_changed = Signal(str, object)
     sig_open_file = Signal(str)
+    sig_delete_project = Signal()

```

Comment:
Mmmh, you're right. @dalthviz, please move the `delete_project` method from `plugins/projects.py` to this widget :-)